crates/wdk-macros/src/lib.rs

```diff
@@ -406,39 +406,53 @@ fn get_wdf_function_info_map(
     types_path: &LitStr,
     span: Span,
 ) -> Result<BTreeMap<String, CachedFunctionInfo>> {
+    #[cfg(test)]
+    let scratch_dir = scratch::path(concat!(env!("CARGO_CRATE_NAME"), "_ast_fragments_test"));
+    #[cfg(not(test))]
     let scratch_dir = scratch::path(concat!(env!("CARGO_CRATE_NAME"), "_ast_fragments"));
-    let flock = std::fs::File::create(scratch_dir.join(".lock"))
-        .to_syn_result(span, "unable to create file")?;
 
     let cached_function_info_map_path = scratch_dir.join("cached_function_info_map.json");
 
     if !cached_function_info_map_path.exists() {
+        let flock = std::fs::File::create(scratch_dir.join(".lock"))
+            .to_syn_result(span, "unable to create file lock")?;
         FileExt::lock_exclusive(&flock).to_syn_result(span, "unable to obtain file lock")?;
 
         // Before this thread acquires the lock, it's possible that a concurrent thread
         // already created the cache. If so, this thread skips cache generation.
         if !cached_function_info_map_path.exists() {
-            let generated_map = create_wdf_function_info_file_cache(
+            let function_info_map = create_wdf_function_info_file_cache(
                 types_path,
                 &cached_function_info_map_path,
                 span,
             )?;
-
             FileExt::unlock(&flock).to_syn_result(span, "unable to unlock file lock")?;
-            return Ok(generated_map);
+            return Ok(function_info_map);
         }
         FileExt::unlock(&flock).to_syn_result(span, "unable to unlock file lock")?;
     }
+    let function_info_map =
+        read_wdf_function_info_file_cache(&cached_function_info_map_path, span)?;
+    Ok(function_info_map)
+}
 
-    let generated_map_string = std::fs::read_to_string(&cached_function_info_map_path)
+/// Reads the cache of function information, then deserializes it into a
+/// `BTreeMap`.
+fn read_wdf_function_info_file_cache(
+    cached_function_info_map_path: &PathBuf,
```

Comment:
```suggestion
    cached_function_info_map_path: &Path,
```
`&Path` is more flexible

Reply:
fixed